gemini-app/src/Components/GCP/ImageViewer.js

```diff
@@ -38,24 +53,104 @@ function ImageViewer({ open, onClose, item, activeTab, platform, sensor }) {
+            location: selectedLocationGCP,
+            population: selectedPopulationGCP,
+            date: selectedDateGCP,
+            platform: selectedPlatformGCP,
+            sensor: selectedSensorGCP,
+        };
+        const gcpLocations = {
+            fields: ["year", "experiment", "location", "population"],
+            fileType: ".csv",
+            label: "GCP Locations"
+        };
+        let dirPath = "";
+        for (const field of gcpLocations.fields) {
+            if (Values[field]) {
+                dirPath += dirPath ? `/${Values[field]}` : Values[field];
+            }
+        }
+        const formData = new FormData();
+        formData.append("files", file);
+        formData.append("dirPath", dirPath);
+        formData.append("dataType", "gcpLocations");
+        const controller = new AbortController();
+        const id = setTimeout(() => controller.abort(), timeout);
+        try {
+            const response = await fetch(`${flaskUrl}upload`, {
+                method: "POST",
+                body: formData,
+                signal: controller.signal,
+            });
+            console.log(response)
+            clearTimeout(id);
+            return response;
+        } catch (error) {
+            console.log("Upload error:", error);
+            clearTimeout(id);
+            throw error;
+        }
+    };
+
+    const fetchAndSetGcpFilePath = async () => {
+        console.log('${flaskUrl', flaskUrl);
+        const files = await fetchData(`${flaskUrl}list_files/Raw/${selectedYearGCP}/${selectedExperimentGCP}/${selectedLocationGCP}/${selectedPopulationGCP}`);
+        const gcpLocationsFile = files.find((file) => file === "gcp_locations.csv");
+        if (gcpLocationsFile) {
+            const newPath = `Raw/${selectedYearGCP}/${selectedExperimentGCP}/${selectedLocationGCP}/${selectedPopulationGCP}/${gcpLocationsFile}`;
+            setPrepGcpFilePath(newPath);
+            console.log("GCP path found, setting to ", newPath);
+        } else {
+            setPrepGcpFilePath("");
+            console.log("No GCP path found");
+        }
+    };
+
+    const handleFileUpload = async (event) => {
+        await uploadFileWithTimeout(event.target.files[0]);
+        await fetchAndSetGcpFilePath();
+        console.log("prepGcpFilePath after setting: ", prepGcpFilePath);
+    };
+
+    const handleDialogClose = () => {
+        setDialogOpen(false);
+        setImageViewerReady(true);
+        setIsImageViewerOpen(true);
+    };
+
     // update state variables
     useEffect(() => {
         if (open) {
+            setImageViewerReady(false);
+            console.log("prepGcpFilePath initial: ", prepGcpFilePath);
             setSelectedDateGCP(item.date);
             setSelectedPlatformGCP(platform);
             setSelectedSensorGCP(sensor);
-            setIsImageViewerOpen(true);
+            //fetchAndSetGcpFilePath();
+            if (prepGcpFilePath != "") {
```

Comment:
For some reason, when the gcp file is already uploaded, it still opens the upload window? Im not sure if that is intended.

<img width="498" alt="Screenshot 2024-12-03 at 10 03 37 AM" src="https://github.com/user-attachments/assets/3457b68d-82c5-4a9f-876b-dfa3f27f8295">

![Screenshot 2024-12-03 at 10 03 44 AM](https://github.com/user-attachments/assets/84599c21-c2d6-4f5d-95b3-4cbd28efa46c)